.neuro/live-pachyderm.yml

```
@@ -333,3 +280,54 @@ jobs:
       server --host 0.0.0.0
         --backend-store-uri=postgresql://postgres:password@${{ inspect_job('postgres').internal_hostname_named }}:5432
         --default-artifact-root=/usr/local/share/mlruns
+
+  create_pipeline:
+    name: $[[ flow.title ]]-create-pipeline
+    image: $[[ images.train.ref ]]
+    volumes:
+      - secret:/$[[ project.owner ]]/gh-rsa:/root/.ssh/id-rsa
+    params:
+      mlflow_storage:
+        descr: Storage path, where MLFlow server stores trained model binaries
+      mlflow_uri:
+        descr: MLFlow server URI
+      pachy_pipeline_name:
+        default: train-dogs
+        descr: MLFlow server URI
+    env:
+      EXPOSE_SSH: "yes"
+      PYTHONPATH: /usr/project
+      PROJECT: /usr/project
+      MLFLOW_STORAGE: ${{ params.mlflow_storage }}
+      MLFLOW_URI: ${{ params.mlflow_uri }}
+      NEURO_PASSED_CONFIG: secret:platform-config
+      TRAIN_IMAGE_REF: ${{ images.train.ref }}
+      PACHY_PIPELINE_NAME: ${{ params.pachy_pipeline_name }}
+    bash: |
+      git clone -b ${PROJECT_GIT_BRANCH} ${PROJECT_GIT_REPO} ${PROJECT}
+
+      # Init pachctl
+      pachctl config update context default --pachd-address ${PACHY_URI}
+      pachctl config set active-context default
+
+      # Store Neuro-flow batch and project definition base-64 encoded
+      BATCH_B64=$(cat ${PROJECT}/.neuro/training.yaml | base64 -w 0)
+      PROJECT_B64=$(cat ${PROJECT}/.neuro/project.yml | base64 -w 0)
+
+      # Substitue Neuro-flow batch parameters from job ENV vars and store them b64 encoded
+      PARAMS_B64=$(cat ${PROJECT}/.neuro/training-batch-params.yaml | envsubst | base64 -w 0)
+
+      # Create a copy of pipeline template
+      cp $${PROJECT}/config/pipeline.json /tmp/pipeline.json
+      sed -i -e s/##NEURO_PASSED_CONFIG##/${NEURO_PASSED_CONFIG}/ /tmp/pipeline.json
+
+      # Probagate project and batch definition and its parameters into the Pachyderm pypeline definition
+      sed -i -e s/##PROJECT_B64##/${PROJECT_B64}/ -e s/##BATCH_B64##/${BATCH_B64}/ -e s/##PARAMS_B64##/${PARAMS_B64}/ /tmp/pipeline.json
+
+      # Set the pipeline name
+      sed -i -e s/##PIPELINE_NAME##/${PACHY_PIPELINE_NAME}/ /tmp/pipeline.json
+
+      # Create pipeline
+      pachctl list pipeline $PACHY_PIPELINE_NAME >/dev/null && pachctl delete pipeline $PACHY_PIPELINE_NAME >/dev/null
+      pachctl create pipeline -f /tmp/pipeline.json
+      echo "Pipeline '${PACHY_PIPELINE_NAME}' created successfully."
```

This fails if the input repo does not exist.
Let's check it and create if not exist.

```suggestion
      # Create input repo
      pachctl inspect repo ${PACHY_REPO} || pachctl create repo ${PACHY_REPO}
      
      # Create pipeline
      pachctl list pipeline $PACHY_PIPELINE_NAME >/dev/null && pachctl delete pipeline $PACHY_PIPELINE_NAME >/dev/null
      pachctl create pipeline -f /tmp/pipeline.json
      echo "Pipeline '${PACHY_PIPELINE_NAME}' created successfully."
```